onnx/defs/tensor/defs.cc

```diff
@@ -697,7 +707,11 @@ ONNX_OPERATOR_SET_SCHEMA(
             ctx.getOutputType(0)
                 ->mutable_tensor_type()
                 ->mutable_shape()
-                ->add_dim();
+                ->add_dim()
+                ->set_dim_value(
+                (i < axis) ? data_shape.dim(i).dim_value() :                             // i < axis < r
```

Comment:
Is dim_value always available?

Reply:
Since `i < axis < r` I think it exists, or did I misunderstand?

Reply:
If you open up the protobuf defs. this field is optional.